dhnx/simulation.py

```diff
@@ -10,23 +10,445 @@
+        # NOTE: pipes have distributed and localized pressure losses zeta_dis * L/D**5
+        # nodes-pressure_losses
+        # NOTE: nodes have only localized pressure losses
+        # 'global-pressure_losses'
+        # 'producers-pump_power'
+
+        def _calculate_reynolds():
+            r"""
+            Calculates the reynolds number.
+
+            ..math::
+
+                Re = \frac{4\dot{m}{\pi\mu D}
+
+            Returns
+            -------
+
+            re : pd.DataFrame
+            """
+            pipes_mass_flow = self.results['pipes-mass_flow']
+
+            diameter = self.thermal_network.components.pipes[['from_node', 'to_node', 'diameter_mm']]
+
+            diameter = 1e-3 * diameter.set_index(['from_node', 'to_node'])['diameter_mm']
+
+            re = 4 * pipes_mass_flow.divide(diameter, axis='columns') \
+                / (np.pi * self.mu)
+
+            return re
+
+        def _calculate_lambda(re):
+            r"""
+            Calculates the darcy friction factor.
+
+            ..math::
+
+                \lambda = 0.007 \cdot Re^{-0.13} \cdot D ^ {-0.14}
+
+            Parameters
+            ----------
+            re
+
+            Returns
+            -------
+
+            lamb : pd.DataFrame
+            """
+
+            factor_diameter = self.thermal_network.components.pipes[
+                ['from_node', 'to_node', 'diameter_mm']
+            ]
+
+            factor_diameter =  (1e-3 \
+                * factor_diameter.set_index(['from_node', 'to_node'])['diameter_mm']) ** -0.14
+
+            lamb = 0.07 * re ** -0.13
+
+            lamb = lamb.multiply(factor_diameter, axis='columns')
+
+            return lamb
+
+        def _calculate_pipes_pressure_losses(lamb):
+            r"""
+            Calculates the pressure losses in the pipes.
+
+            .. math::
+
+            \delta p = \lambda \frac{8L}{\rho \pi^2 D^5}\dot{m}^2.
+
+            Parameters
+            ----------
+            lamb : pd.DataFrame
+                DataFrame with lambda values for each timestep.
+
+            Returns
+            -------
+            pipes_pressure_losses : pd.DataFrame
+                DataFrame with pressure losses values for each timestep.
+            """
+            pipes_mass_flow = self.results['pipes-mass_flow'].copy()
+
+            constant = 8 * lamb / (self.rho * np.pi**2)
+
+            length = self.thermal_network.components.pipes[['from_node', 'to_node', 'length_m']]
+
+            diameter = self.thermal_network.components.pipes[['from_node', 'to_node', 'length_m']]
+
+            length = length.set_index(['from_node', 'to_node'])['length_m']
+
+            diameter = diameter.set_index(['from_node', 'to_node'])['length_m']
+
+            diameter_5 = 1e-3 * diameter ** 5
+
+            pipes_pressure_losses = constant * pipes_mass_flow\
+                .multiply(length, axis='columns')\
+                .divide(diameter_5, axis='columns')
```

Comment:
```suggestion
            pipes_pressure_losses = constant * pipes_mass_flow\
                .multiply(pipes_mass_flow, axis='columns')\
                .multiply(length, axis='columns')\
                .divide(diameter_5, axis='columns')
```

It's squared mass flow in this equation. Is there a more elegant way than multiplying it with itself?

Reply:
I squared it before multiplying (2d19cdc .